src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs

```diff
@@ -29,32 +31,35 @@ private enum OperationKind
             RenameFile
         }
 
-        protected virtual SyntaxNode GetNodeToAnalyze(SyntaxNode root, TextSpan span) => root.FindNode(span);
-
-        public async Task<ImmutableArray<CodeAction>> GetRefactoringAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken)
+        public async Task<ImmutableArray<CodeAction>> GetRefactoringAsync(
+            Document document, TextSpan textSpan, CancellationToken cancellationToken)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            if (!ShouldAnalyze(root, textSpan))
-            {
-                return default(ImmutableArray<CodeAction>);
-            }
-
-            var semanticDocument = await SemanticDocument.CreateAsync(document, cancellationToken).ConfigureAwait(false);
-            var state = State.Generate((TService)this, semanticDocument, textSpan, cancellationToken);
-            if (state == null)
+            if (textSpan.IsEmpty)
```

Comment:
Why the change to only do it on empty selections?  Many people will double click a name and expect something like this to work.